src/domains/citizen-auth-app/10_function_public.tf

```diff
@@ -0,0 +1,258 @@
+
+      VALIDATION_CALLBACK_URL = "https://api-app.io.pagopa.it/email_verification.html"
+      CONFIRM_CHOICE_PAGE_URL = "https://api-app.io.pagopa.it/email_confirm.html"
+    }
+  }
+}
+
+resource "azurerm_resource_group" "public_rg_itn" {
+  name     = format("%s-public-rg-01", local.short_project_itn)
+  location = local.itn_location
+
+  tags = var.tags
+}
+
+
+module "function_public_snet_itn" {
+  source                                    = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v8.44.0"
+  name                                      = format("%s-public-snet-01", local.short_project_itn)
+  address_prefixes                          = var.cidr_subnet_fnpublic_itn
+  resource_group_name                       = data.azurerm_virtual_network.common_vnet_italy_north.resource_group_name
+  virtual_network_name                      = data.azurerm_virtual_network.common_vnet_italy_north.name
+  private_endpoint_network_policies_enabled = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+    "Microsoft.AzureCosmosDB",
+    "Microsoft.Storage",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "function_public_itn" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//function_app?ref=v8.44.0"
+
+  resource_group_name = azurerm_resource_group.public_rg_itn.name
+  name                = format("%s-public-func-01", local.short_project_itn)
+  location            = local.itn_location
+  domain              = "auth"
+  health_check_path   = "/info"
+
+  node_version    = "18"
+  runtime_version = "~4"
+
+  app_service_plan_info = {
+    kind                         = var.function_public_kind
+    sku_size                     = var.function_public_sku_size
+    maximum_elastic_worker_count = 0
+    worker_count                 = null
+    zone_balancing_enabled       = true
+  }
+
+  always_on                                = "true"
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_settings = merge(
+    local.function_public.app_settings_common,
+  )
+
+  internal_storage = {
+    "enable"                     = false,
+    "private_endpoint_subnet_id" = data.azurerm_subnet.itn_pep.id,
+    "private_dns_zone_blob_ids"  = [data.azurerm_private_dns_zone.privatelink_blob_core_windows_net.id],
+    "private_dns_zone_queue_ids" = [data.azurerm_private_dns_zone.privatelink_queue_core_windows_net.id],
+    "private_dns_zone_table_ids" = [data.azurerm_private_dns_zone.privatelink_table_core_windows_net.id],
+    "queues"                     = [],
+    "containers"                 = [],
+    "blobs_retention_days"       = 0,
+  }
+
+  subnet_id = module.function_public_snet_itn.id
+
+  allowed_subnets = [
```

Comment:
We've been using peps, so this property should be null and `enable_function_app_public_network_access` must be set to `false`.

However, as we've seen last time this condition is not true when DevOps is being used. If this is the case, set `data.azurerm_subnet.apim_v2_snet.id` as the only allowed subnet and do not set `enable_function_app_public_network_access` variable

Reply:
thank you @Krusty93. we are still using azdevops here, I removed the unnecessary in https://github.com/pagopa/io-infra/pull/1244/commits/8feb975ccc66152b87932e21eb9a2ef89b5ee2eb

Reply:
I think we can safely remove the Apim submit, anyway. 
The only one we still need is `data.azurerm_subnet.azdoa_snet[0].id` for staging.

Reply:
Great

Reply:
done in https://github.com/pagopa/io-infra/pull/1244/commits/020325834ce684b2025231e96902ff9e247579ea